src/Pelagos/Bundle/AppBundle/Form/DatasetSubmissionType.php

```diff
@@ -283,6 +299,14 @@ public function buildForm(FormBuilderInterface $builder, array $options)
                 'label' => 'Request Revisions',
                 'attr'  => array('class' => 'submitButton'),
             ));
+
+        $builder->addEventListener(FormEvents::POST_SET_DATA, function (FormEvent $event) {
+            $data = $event->getData();
+            $form = $event->getForm();
+            if ($data->isDatasetFileInColdStorage() === true) {
```

Comment:
I think this function is an overkill? You have client side and server side validation in place

Reply:
Yes, It's kinda overkill but this way works fine and it also guarantees that everything is right from the backend IMO